src/renderer/components/dialog-add-version.tsx

```diff
@@ -168,25 +195,33 @@ export class AddVersionDialog extends React.Component<
   }
 
   private renderPath(): JSX.Element | null {
-    const { folderPath, isValidElectron } = this.state;
+    const { isValidElectron, folderPath, existingLocalVersion } = this.state;
+    const shouldSetToPreviousVersion = isValidElectron && existingLocalVersion;
 
     if (!folderPath) return null;
-
-    const info = isValidElectron
-      ? `We found an ${getElectronNameForPlatform()} in this folder.`
-      : `We did not find a ${getElectronNameForPlatform()} in this folder...`;
-
     return (
       <Callout>
-        {info}
-        {this.renderVersionInput()}
+        {this.getVersionFromPathDialog()}
+        {!shouldSetToPreviousVersion && this.renderVersionInput()}
       </Callout>
     );
   }
 
+  private getVersionFromPathDialog(): string {
```

Comment:
This function name is slightly misleading; it implies a version will be returned but it's a message.

Maybe something like `getConfirmationMessage()` or `buildDialogText()`?